### PR TITLE
Fix to add built-in labels into Cloud Run resources when using promote

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nghialv** [comment]
@knanao btw, I think this must be `e.Deployment.RunningCommitHash`

https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/executor/cloudrun/rollback.go#L101

**Review by knanao — COMMENTED** [review]
@nghialv
Ops, I misunderstood it. let me fix it.

**Comment by knanao** [comment]
@nghialv @khanhtc1202 
I fixed some points, PTAL when you guys have enough time.

File: pkg/app/piped/executor/cloudrun/cloudrun.go [hunk 1]
```diff
@@ -158,6 +158,9 @@ func addBuiltinLabels(sm provider.ServiceManifest, hash, pipedID, appID, revisio
 	// Set builtinLabels for Service.
 	sm.AddLabels(labels)
 
+	if revisionName == "" {
+		return true
+	}
```
Comment (nghialv) [inline pkg/app/piped/executor/cloudrun/cloudrun.go:163]: I don't understand why do we need this?
If the `revisionName` is empty, the `sm.AddRevisionLabels(labels)` is not called, is that ok?

Comment (knanao) [reply pkg/app/piped/executor/cloudrun/cloudrun.go:163]: If the revision have already existed, newRevision will be empty and this mean that needn't to set label again.
However, we can fix like this.

```
if newRevision != "" {
  if !addBuiltinLabels(sm, commit, e.PipedConfig.PipedID, e.Deployment.ApplicationId, newRevision, e.LogPersister) {
           return model.StageStatus_STAGE_FAILURE
  }
}
```
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/executor/cloudrun/deploy.go#L189-L192

Comment (nghialv) [reply pkg/app/piped/executor/cloudrun/cloudrun.go:163]: I got it. So let's check it's empty at the caller side as you commented above.

Comment (knanao) [reply pkg/app/piped/executor/cloudrun/cloudrun.go]: Allright, I addressed https://github.com/pipe-cd/pipecd/pull/3438/commits/c1550eb73f5d2e8c0fd45440f17a5cce99ea1b81

**Comment by nghialv** [comment]
Thank you.

/lgtm

**Comment by khanhtc1202** [comment]
Good fix
/approve

**Pull request merged**